webapps-backup/src/main/java/io/camunda/webapps/backup/repository/opensearch/OpenSearchSnapshotInfo.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.webapps.backup.repository.opensearch;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.opensearch.client.json.JsonData;
+import org.opensearch.client.opensearch.snapshot.SnapshotInfo;
+import org.opensearch.client.opensearch.snapshot.SnapshotShardFailure;
+
+public class OpenSearchSnapshotInfo {
+
+  private String snapshot;
+  private String uuid;
+
+  private SnapshotState state;
+
+  private List<SnapshotShardFailure> failures = List.of();
+
+  private Long startTimeInMillis;
+
+  private Long endTimeInMillis;
+
+  private Map<String, JsonData> metadata = Map.of();
+
+  public String getSnapshot() {
+    return snapshot;
+  }
+
+  public OpenSearchSnapshotInfo setSnapshot(final String snapshot) {
+    this.snapshot = snapshot;
+    return this;
+  }
+
+  public String getUuid() {
+    return uuid;
+  }
+
+  public OpenSearchSnapshotInfo setUuid(final String uuid) {
+    this.uuid = uuid;
+    return this;
+  }
+
+  public SnapshotState getState() {
+    return state;
+  }
+
+  public OpenSearchSnapshotInfo setState(final SnapshotState state) {
+    this.state = state;
+    return this;
+  }
+
+  public List<SnapshotShardFailure> getFailures() {
+    return failures;
+  }
+
+  public OpenSearchSnapshotInfo setFailures(final List<SnapshotShardFailure> failures) {
+    this.failures = failures;
+    return this;
+  }
+
+  public Long getStartTimeInMillis() {
+    return startTimeInMillis;
+  }
+
+  public OpenSearchSnapshotInfo setStartTimeInMillis(final Long startTimeInMillis) {
+    this.startTimeInMillis = startTimeInMillis;
+    return this;
+  }
+
+  public Long getEndTimeInMillis() {
+    return endTimeInMillis;
+  }
+
+  public OpenSearchSnapshotInfo setEndTimeInMillis(final Long endTimeInMillis) {
+    this.endTimeInMillis = endTimeInMillis;
+    return this;
+  }
+
+  public Map<String, JsonData> getMetadata() {
+    return metadata;
+  }
+
+  public OpenSearchSnapshotInfo setMetadata(final Map<String, JsonData> metadata) {
+    this.metadata = metadata;
+    return this;
+  }
+
+  public static OpenSearchSnapshotInfo fromResponse(final SnapshotInfo snapshotInfo) {
+    return new OpenSearchSnapshotInfo()
```

Comment:
factory copied from `OpenSearchSnapshotOperations` except it was using an untyped map, while here we are using the typed response from the client.